### PR TITLE
[EAP7-881]: Provide launch.openshift.io experience

### PR DESCRIPTION
**Pull request by ehsavoie**
* Add support for new application creation on launch.openshift.io
* Add support for missions on launch.openshift.io

Jira: https://issues.jboss.org/browse/EAP7-881

### CODE REVIEW

**Comment by jmesnil** [comment]
@ehsavoie can you move the file under `cloud/` please?

File: EAP7-881.adoc [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+We need to provide the following capabilities of the launcher creator:
+
+- database: to support connection and persistence on a remote database (currently MySQL and Postgresql are supported)
+- health: to provide an health check endpoint.
+- rest: to provide a REST HelloWorld API.
+ 
+Following the document https://github.com/fabric8-launcher/launcher-creator-backend/wiki/Contributing  we are going to provide the generators to support those capabilities:
+
+* create a *platform-wildfly* generator to define the base configuration of a WildFly application with a health check endpoint on _/health_ returning a simple OK.
+Since this must be exposed we can't use the Microprofile Health support of WildFly which is on 9990.
```
Comment (bstansberry) [inline EAP7-881.adoc]: What is this then? Some custom user created webapp that provides their own 'health' context? Why would we teach people to do that?

File: EAP7-881.adoc [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+Here we are also configuring the generated YAML for the OpenShift deployment support.
+* create a *rest-wildfly* generator to define a really simple greeting API: we need to provide a REST endpoint on _/api/greeting_ with a name parameter and returning a simple JSON text.
+* create a *database-crud-wildfly* generator to define a really fruit list managing API, persisting the changes in a database. Here we will need to configure the server image to connect to a remote database using OpenShift secrets.
+
+=== Deploy an Example Application
+
+There are currently several *missions* defined in a catalog for each runtime: https://github.com/fabric8-launcher/launcher-documentation/wiki. We will need to select and implement those missions for WildFly / EAP :
```
Comment (bstansberry) [inline EAP7-881.adoc]: So we choose which ones we wish to do?

File: EAP7-881.adoc [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+- health: to provide an health check endpoint.
+- rest: to provide a REST HelloWorld API.
+ 
+Following the document https://github.com/fabric8-launcher/launcher-creator-backend/wiki/Contributing  we are going to provide the generators to support those capabilities:
+
+* create a *platform-wildfly* generator to define the base configuration of a WildFly application with a health check endpoint on _/health_ returning a simple OK.
+Since this must be exposed we can't use the Microprofile Health support of WildFly which is on 9990.
+Here we are also configuring the generated YAML for the OpenShift deployment support.
+* create a *rest-wildfly* generator to define a really simple greeting API: we need to provide a REST endpoint on _/api/greeting_ with a name parameter and returning a simple JSON text.
+* create a *database-crud-wildfly* generator to define a really fruit list managing API, persisting the changes in a database. Here we will need to configure the server image to connect to a remote database using OpenShift secrets.
```
Comment (bstansberry) [inline EAP7-881.adoc]: Our images have mechanisms for doing database integration, and those will likely evolve over the course of the next six months. So how the launcher works will need to be part of an overall strategy for doing this.

File: EAP7-881.adoc (hunk 1 above)
Comment (ehsavoie) [reply EAP7-881.adoc]: This is used by the launcher UI

File: EAP7-881.adoc (hunk 3 above)
Comment (ehsavoie) [reply EAP7-881.adoc]: we will need to upgrade the generators to follow those changes. Also we can control the version of the images we are using.

File: EAP7-881.adoc (hunk 2 above)
Comment (ehsavoie) [reply EAP7-881.adoc]: Well, all those that are starter are not required :) Then we have to implement those we can/want.

File: EAP7-881.adoc [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-881
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
```
Comment (nziakova) [inline EAP7-881.adoc]: This will be me - nziakova@redhat.com

File: EAP7-881.adoc [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+=== Affected Projects or Components
+
+This support is provided by Fabric8 launcher: https://github.com/fabric8-launcher.
+
+=== Other Interested Projects
+
+== Requirements
+
+ * provide generators for creation of new applications with WildFly
```
Comment (nziakova) [inline EAP7-881.adoc]: I can see WildFly option at https://launch.openshift.io/launch - is the WF part done already or does this cover something else?

File: EAP7-881.adoc [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+* 109: Istio Distributed Tracing
+* 110: Istio Circuit Breaker
+* 111: Istio Routing
+* 200: AMQ Messaging Work Queue Mission
+
+== Test Plan
+
+== Community Documentation
+////
+Generally a feature should have documentation as part of the PR to wildfly master, or as a follow up PR if the feature is in wildfly-core. In some cases though the documentation belongs more in a component, or does not need any documentation. Indicate which of these will happen.
```
Comment (nziakova) [inline EAP7-881.adoc]: Will this need any documentation on WFLY/EAP side? Or will only https://github.com/fabric8-launcher/launcher-documentation/wiki be updated?

**Comment by ehsavoie** [comment]
OUTDATED

**Pull request closed without merging**